storages/backends/s3boto.py

```diff
@@ -494,3 +495,9 @@ def get_available_name(self, name, max_length=None):
             name = self._clean_name(name)
             return name
         return super(S3BotoStorage, self).get_available_name(name, max_length)
+
+    def update_headers(self, headers, name=None, content=None):
+        """
+        Method that can be used to adjust file parameters.
+        """
+        return headers
```

Comment:
Probably we can move headers update functionality from _save() into this method

```
    def update_headers(self, name=None, content=None, content_type=None):
        """
        Method that can be used to adjust file parameters.
        """
        headers = self.headers.copy()

        # setting the content_type in the key object is not enough.
        headers.update({'Content-Type': content_type})
 
        if self.gzip and content_type in self.gzip_content_types:
            headers.update({'Content-Encoding': 'gzip'})
        elif encoding:
            # If the content already has a particular encoding, set it
            headers.update({'Content-Encoding': encoding})

        return headers
```

Reply:
yes. this approach would make `save` cleaner. I'm conviced.